src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.DefaultArrayPoolEvents",
```

Comment:
Nit: I wonder if we should name this more generally, allowing it to be the event source that's used for any future eventing done from anywhere in the assembly, e.g. System.Buffers.BuffersEventSource, or something like that.  I believe there are overheads associated with having lots of event source instances and thus a desire to keep the number of them down to a reasonable count.  That might be out-of-date info, though.